test suite requires deprecated GODEBUG=tls13=0 to pass

<strike>This unit test fails consistently when using go 1.13. As we will plan to upgrade to go 1.13 in the future, we should no doubt fix it:</strike>

The unit test currently has a workaround set in the Makefile, of `export GODEBUG=tls13=0`. This is require for the unit tests to not produce the following error.  According to the [go docs](https://golang.org/doc/go1.13#tls_1_3), this workaround will be removed in go 1.14.

```
ok  	vitess.io/vitess/go/memcache	0.002s
E1208 20:51:09.808962    6454 server.go:320] Cannot read post-SSL client handshake response from client 1 ([::1]:36288): tls: client didn't provide a certificate
io.ReadFull(header size) failed
W1208 20:51:09.809726    6454 auth_server_static.go:195] Config parsed using legacy configuration. Please update to the latest format: {"user":[{"Password": "xxx"}, ...]}
W1208 20:51:10.991662    6454 server.go:363] Error authenticating user using MySQL native password: Access denied for user 'user1' (errno 1045) (sqlstate 28000)
W1208 20:51:10.992049    6454 server.go:363] Error authenticating user using MySQL native password: Access denied for user 'user1' (errno 1045) (sqlstate 28000)
W1208 20:51:11.005966    6454 server.go:451] Slow connection from client 1 (127.0.0.1:41664): 446.373µs
E1208 20:51:11.014685    6454 server.go:273] mysql_server caught panic:
test panic attack!
/home/morgo/vitess/go/mysql/server_test.go:93 (0x8a5e97)
	io/vitess/go/mysql.(*testHandler).ComQuery: panic("test panic attack!")
/home/morgo/vitess/go/mysql/conn.go:1070 (0x8465f7)
	io/vitess/go/mysql.(*Conn).execQuery: err := handler.ComQuery(c, query, func(qr *sqltypes.Result) error {
/home/morgo/vitess/go/mysql/conn.go:771 (0x845e6d)
	io/vitess/go/mysql.(*Conn).handleNextCommand: if err := c.execQuery(sql, handler, more); err != nil {
/home/morgo/vitess/go/mysql/server.go:458 (0x863dfd)
	io/vitess/go/mysql.(*Listener).handle: err := c.handleNextCommand(l.handler)
/usr/local/go/src/runtime/asm_amd64.s:1357 (0x460c00)
	goexit: BYTE	$0x90	// NOP
E1208 20:51:11.034205    6454 conn.go:1117] Error in the middle of a stream to client 5 (127.0.0.1:41672): forced error after send (errno 1047) (sqlstate 08S01)
W1208 20:51:11.063186    6454 server.go:395] Error authenticating user using MySQL native password: Access denied for user 'user1' (errno 1045) (sqlstate 28000)
W1208 20:51:11.068802    6454 server.go:395] Error authenticating user using MySQL native password: Access denied for user 'user2' (errno 1045) (sqlstate 28000)
--- FAIL: TestTLSServer (0.23s)
    server_test.go:1043: No count found for version TLS12
FAIL
FAIL	vitess.io/vitess/go/mysql	2.349s
ok  	vitess.io/vitess/go/mysql/ldapauthserver	0.003s
```

The part of the log that tells the actual error is:

```
--- FAIL: TestTLSServer (0.23s)
    server_test.go:1043: No count found for version TLS12
```

It might be as simple as the fact that Go 1.13 changes the default TLS version to 1.3:

https://golang.org/doc/go1.13#tls_1_3

So we probably just need to change the test to accept either 1.2 or 1.3, to maintain compatibility with older Go versions.

--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,6 @@ MAKEFLAGS = -s
 
 export GOBIN=$(PWD)/bin
 export GO111MODULE=on
-export GODEBUG=tls13=0
 export REWRITER=go/vt/sqlparser/rewriter.go
 
 # Disabled parallel processing of target prerequisites to avoid that integration tests are racing each other (e.g. for ports) and may fail.

--- a/go/mysql/server.go
+++ b/go/mysql/server.go
@@ -47,6 +47,7 @@ const (
 	versionTLS10      = "TLS10"
 	versionTLS11      = "TLS11"
 	versionTLS12      = "TLS12"
+	versionTLS13      = "TLS13"
 	versionTLSUnknown = "UnknownTLSVersion"
 	versionNoTLS      = "None"
 )
@@ -798,6 +799,8 @@ func tlsVersionToString(version uint16) string {
 		return versionTLS11
 	case tls.VersionTLS12:
 		return versionTLS12
+	case tls.VersionTLS13:
+		return versionTLS13
 	default:
 		return versionTLSUnknown
 	}

--- a/go/mysql/server_test.go
+++ b/go/mysql/server_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package mysql
 
 import (
+	"crypto/tls"
 	"fmt"
 	"io/ioutil"
 	"net"
@@ -1006,7 +1007,11 @@ func TestTLSServer(t *testing.T) {
 		t.Errorf("Unexpected output for 'ssl echo': %v", results)
 	}
 
-	checkCountForTLSVer(t, versionTLS12, 1)
+	// Find out which TLS version the connection actually used,
+	// so we can check that the corresponding counter was incremented.
+	tlsVersion := conn.conn.(*tls.Conn).ConnectionState().Version
+
+	checkCountForTLSVer(t, tlsVersionToString(tlsVersion), 1)
 	checkCountForTLSVer(t, versionNoTLS, 0)
 	conn.Close()
 
@@ -1101,7 +1106,7 @@ func checkCountForTLSVer(t *testing.T, version string, expected int64) {
 			t.Errorf("Expected connection count for version %s to be %d, got %d", version, expected, count)
 		}
 	} else {
-		t.Errorf("No count found for version %s", version)
+		t.Errorf("No count for version %s found in %v", version, connCounts)
 	}
 }
 

